packages/core/src/createTheme.js

```diff
@@ -0,0 +1,340 @@
+import defaultTheme from './theme'
+const deepmerge = require('deepmerge')
+
+/**
+ * Create the palette based on passed in theme
+ *
+ * @param {Object=} palette The passed in palette
+ * @param {Object}  theme   The passed in theme
+ *
+ * @returns {Object} The generated palette
+ */
+const createPalette = ({ palette = {}, ...theme }) => {
+  const {
+    lightBlue,
+    blue,
+    darkBlue,
+    lightGreen,
+    green,
+    darkGreen,
+    white,
+    text,
+    lightRed,
+    red,
+    darkRed,
+    lightOrange,
+    orange,
+    darkOrange,
+    lightYellow,
+    yellow,
+    lightPurple,
+    purple,
+    pink,
+    gray,
+    borderGray,
+    darkGray,
+    lightGray,
+    buttonGray
+  } = theme.colors
+
+  return deepmerge(
+    {
+      primary: {
+        light: lightBlue,
+        base: blue,
+        dark: darkBlue
+      },
+      secondary: {
+        light: lightGreen,
+        base: green,
+        dark: darkGreen
+      },
+      text: {
+        light: white,
+        base: text,
+        dark: '#001023'
+      },
+      success: {
+        light: lightGreen,
+        base: green,
+        dark: darkGreen
+      },
+      error: {
+        light: lightRed,
+        base: red,
+        dark: darkRed
+      },
+      warning: {
+        light: lightRed,
+        base: red,
+        dark: darkRed
+      },
+      alert: {
+        light: lightOrange,
+        base: orange,
+        dark: darkOrange
+      },
+      caution: {
+        light: lightYellow,
+        base: yellow,
+        dark: '#b19a1d'
+      },
+      notify: {
+        light: lightYellow,
+        base: yellow,
+        dark: '#b19a1d'
+      },
+      pricePrimary: {
+        light: lightGreen,
+        base: green,
+        dark: darkGreen
+      },
+      priceSecondary: {
+        light: blue,
+        base: darkBlue,
+        dark: '#002f6b'
+      },
+      promoPrimary: {
+        light: lightPurple,
+        base: purple,
+        dark: '#550080'
+      },
+      promoSecondary: {
+        light: '#fe649a',
```

Comment:
Should these hex values have semantic color names for improved readability?

Reply:
Eventually all of the old `colors.[name]` will be removed and we will only have this object with hex values.